internal/cmd/server.go

```diff
@@ -49,19 +67,82 @@ func server() error {
 		eg, err := config.Decode(cfgPath)
 		if err != nil {
 			log.Error(err, "failed to decode config file", "name", cfgPath)
-			os.Exit(1)
+			return nil, err
 		}
 		// Set defaults for unset fields
 		eg.SetDefaults()
 		cfg.EnvoyGateway = eg
 	}
+	return cfg, nil
+}
+
+// setupServices starts all the services required for the Envoy Gateway to
+// fulfill its tasks.
+func setupServices(cfg *config.Server) error {
+	// TODO - Setup a Config Manager
+	// https://github.com/envoyproxy/gateway/issues/43
+	ctx := ctrl.SetupSignalHandler()
 
-	k8sTable := new(provider.ResourceTable)
-	if err := provider.Start(cfg, k8sTable); err != nil {
+	pResources := new(message.ProviderResources)
+	// Start the Provider Service
+	// It fetches the resources from the configured provider type
+	// and publishes it
+	providerSvc := &providersvc.Service{
```

Comment:
`providersvc.Service` studders. Should we make `Service` as part of the component pkg so we have `provider.Service`?

Reply:
renamed to Runner with a New method